lib/fathead/spring_framework/parse_utils.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python
+from bs4 import BeautifulSoup
+import os
+import re
+
+BASE_JAVADOC_URL = "http://docs.spring.io/spring/docs/current/javadoc-api/"
+BASE_LOCAL_JAVADOC_DIR = "./docs/javadoc-api"
+BASE_JAVADOC_FILE = BASE_LOCAL_JAVADOC_DIR + "/allclasses-noframe.html"
+
+
+def readRootFile():
+    rootFile = open(BASE_JAVADOC_FILE, 'r')
+    lines = rootFile.read()
+    rootFile.close()
+    return lines
+
+
+def collectDocFilesFrom(dir):
+    docFiles = []
+    for (path, dirs, files) in os.walk(dir):
+        if 'class-use' not in path and 'index-files' not in path:
+            for f in files:
+                if f.endswith('.html') and 'package-' not in f and 'doc-files' not in f:
+                    docFiles.append("%s/%s" % (path, f))
+    return docFiles
+
+
+def getDocs(filename, classUrl):
+    if filename.endswith('.html') and 'package-' not in filename and 'doc-files' not in filename:
+        content = BeautifulSoup(getcontent(filename), 'html.parser')
+        classname = content.find_all('h2')[0].string
+        block = content.find_all('div', 'block', limit=1)
+        description = ""
+        if len(block) > 0:
+            description = block[0].get_text()
+            description = cutlength(description)
+        url = ""
+        if len(classUrl) != 0:
+            url = BASE_JAVADOC_URL + classUrl
+        return classname, description, url
+
+
+def cutlength(description):
+    #  if len(description) > 100:
+    description = description[0:description.rfind('.', 0, 300) + 1]
```

Comment:
@mandekisd how long to the descriptions get? We automatically suppress the content so that it can't overflow the page, if the majority of the descriptions are only a few paragraphs I don't think we need to limit ourself.

However, there's plenty of other useful information we may want to show instead (e.g. list out the methods, or show example usage, etc)